src/kayenta/report/list/table.tsx

```diff
@@ -179,25 +195,64 @@ const TableRows = ({ executions }: { executions: ICanaryExecutionStatusResult[]
   );
 };
 
-const ExecutionListTable = ({ executions, application, accounts }: IExecutionListTableStateProps) => (
-  <div className="vertical execution-list-container">
-    {CanarySettings.manualAnalysisEnabled && (
-      <button
-        style={{ alignSelf: 'flex-end', flexShrink: 0 }}
-        className="primary"
-        onClick={() => startManualAnalysis(application, accounts)}
-      >
-        <i className="fa fa-play" /> Start Manual Analysis
-      </button>
-    )}
-    <TableRows executions={executions} />
-  </div>
-);
+const updateExecutionsCount = (event: React.ChangeEvent<HTMLSelectElement>) => {
+  ReactInjector.$state.go('.', { count: Number(event.target.value) });
+};
+
+const ExecutionListTable = ({ executions, application, accounts, executionsCount }: IExecutionListTableStateProps) => {
+  React.useEffect(() => {
+    const dataSource = application.getDataSource('canaryExecutions');
+    if (!dataSource.active) {
+      dataSource.activate();
+    }
+    dataSource.refresh();
+    return () => dataSource.deactivate();
+  }, [application, executionsCount]);
+
+  const countOptions: number[] = CanarySettings.executionsCountOptions ?? [20, 50, 100, 200];
+  if (!countOptions.includes(executionsCount)) {
```

Comment:
if someone is clever and hacks the URL, we'll add what they put as an option in the dropdown